OpenRA.Mods.Common/Widgets/ProductionPaletteWidget.cs

```diff
@@ -383,24 +399,24 @@ bool HandleRightClick(ProductionItem item, ProductionIcon icon, int handleCount)
 				TextNotificationsManager.AddTransientLine(World.LocalPlayer, CurrentQueue.Info.OnHoldTextNotification);
 
 				World.IssueOrder(Order.PauseProduction(CurrentQueue.Actor, icon.Name, true));
+				return true;
 			}
-
-			return true;
```

Comment:
Moving this into the if/else above appears to be a leftover from earlier changes and can now be removed.